eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -377,8 +378,9 @@ object MultiPartPaymentLifecycle {
     SendPaymentToRoute(replyTo, Right(route), finalPayload)
   }
 
-  /** When we receive an error from the final recipient, we should fail the whole payment, it's useless to retry. */
-  private def isFinalRecipientFailure(pf: PaymentFailed, d: PaymentProgress): Boolean = pf.failures.collectFirst {
+  /** When we receive an error from the final recipient or payment gets confirmed on chain, we should fail the whole payment, it's useless to retry. */
+  private def doNotRetry(pf: PaymentFailed, d: PaymentProgress): Boolean = pf.failures.collectFirst {
+    case LocalFailure(_, _: HtlcsTimedoutDownstream | _: HtlcOverriddenByLocalCommit) => true
```

Comment:
This is `HtlcsWillTimeoutUpstream` as well.
You should probably create a dedicated method `isOnChainFail(f: PaymentFailure)`.

Reply:
Seems I'd have to do exactly the same matching in a new method, what about keeping as is but expanding each case: https://github.com/ACINQ/eclair/pull/1799/commits/776d9dfc92906ca665c37fd12954fb58290b487c#diff-f28a1f15ee9515a9b2bb404c65cb4a01415b12d264f509420bdfb1251fd49afbR382-R388 ?